services/app-web/src/pages/StateSubmission/SubmissionType/SubmissionType.tsx

```diff
@@ -87,28 +89,33 @@ export const SubmissionType = ({
             // without a refresh. Anytime a mutation does more than "modify an existing object"
             // you'll need to handle the cache.
             update(cache, { data }) {
-                if (data) {
-                    cache.modify({
-                        fields: {
-                            indexHealthPlanPackages(
-                                index = { totalCount: 0, edges: [] }
-                            ) {
-                                const newID = cache.identify(
-                                    data.createHealthPlanPackage.pkg
-                                )
-                                // This isn't quite what is documented, but it's clear this
-                                // is how things work from looking at the dev-tools
-                                const newRef = { __ref: newID }
+                const pkg = data?.createHealthPlanPackage.pkg
+                if (pkg) {
+                    const result =
+                        cache.readQuery<IndexHealthPlanPackagesQuery>({
+                            query: IndexHealthPlanPackagesDocument,
+                        })
 
-                                return {
-                                    totalCount: index.totalCount + 1,
-                                    edges: [
-                                        {
-                                            node: newRef,
-                                        },
-                                        ...index.edges,
-                                    ],
-                                }
+                    const indexHealthPlanPackages = {
+                        totalCount:
+                            result?.indexHealthPlanPackages.totalCount || 0,
+                        edges: result?.indexHealthPlanPackages.edges || [],
+                    }
+
+                    cache.writeQuery({
+                        query: IndexHealthPlanPackagesDocument,
+                        data: {
+                            indexHealthPlanPackages: {
+                                __typename: 'IndexHealthPlanPackagesPayload',
+                                totalCount:
+                                    indexHealthPlanPackages.totalCount + 1,
+                                edges: [
+                                    {
+                                        __typename: 'HealthPlanPackageEdge',
+                                        node: pkg,
+                                    },
+                                    ...indexHealthPlanPackages.edges,
+                                ],
```

Comment:
💡  I would like to see a technical design doc about what our cacheing approach is since we are integrating it across the app. I feel like there are decisions that have been made or implemented that are not yet reflected in docs. For example, when are using `readQuery` `writeQuery` and when are we `refetchQueries`? 

 We could link [this](https://www.apollographql.com/docs/react/v2/caching/cache-interaction/#updating-after-a-mutation) in the docs. I think this sprint could be a good time to address while its fresh since it also came up last sprint.